### PR TITLE
Rename IntList to IntListRef

### PR DESCRIPTION
**Pull request by ezyang**
Stack:
&nbsp;&nbsp;&nbsp;&nbsp;:white_circle:&nbsp; #16504 Move Deprecated.h to c10&nbsp;&nbsp;[:green_heart:](https://our.internmc.facebook.com/intern/diff/D13860570/)
&nbsp;&nbsp;&nbsp;&nbsp;:black_circle:&nbsp; **#16469 Rename IntList to IntListRef**&nbsp;&nbsp;[:yellow_heart:](https://our.internmc.facebook.com/intern/diff/D13849956/)

- Global search replace IntList to IntListRef
- Deprecated alias for IntList. Well, I'd like to deprecated it, but there seem
  to be build problems doing so.

Differential Revision: [D13849956](https://our.internmc.facebook.com/intern/diff/D13849956/)

### CODE REVIEW

**Comment by ezyang** [comment]
Greg thinks it should be IntList because it's a more Python oriented naming :P

**Comment by gchanan** [comment]
Actually, I take that back because we don't usually expose this name to python (it's `int[]` now).  So `IntArrayRef` or `Int64ArrayRef` works.

File: aten/src/ATen/Declarations.cwrap [hunk 1]
```diff
@@ -131,7 +131,7 @@
   return: THTensor*
   arguments:
     - THTensor* self
-    - arg: IntListSize size
```
Comment (smessmer) [inline aten/src/ATen/Declarations.cwrap]: Is it intentional that you also changed `IntListSize` to `IntListRefSize`?

File: aten/src/ATen/core/ivalue.h [hunk 2]
```diff
@@ -278,22 +278,22 @@ struct CAFFE2_API IValue final {
...
+  IValue(c10::intrusive_ptr<ivalue::IntListRef> v);
   IValue(std::vector<int64_t> v);
   IValue(at::ArrayRef<int64_t> v)
   : IValue(v.vec()) {}
-  bool isIntList() const { return Tag::IntList == tag; }
-  c10::intrusive_ptr<ivalue::IntList> toIntList() && {
-    AT_ASSERT(isIntList());
-    return moveToIntrusivePtr<ivalue::IntList>();
+  bool isIntListRef() const { return Tag::IntListRef == tag; }
+  c10::intrusive_ptr<ivalue::IntListRef> toIntListRef() && {
```
Comment (smessmer) [inline aten/src/ATen/core/ivalue.h]: It should probably stay `toIntList()`

File: aten/src/ATen/core/ivalue.h [hunk 3]
```diff
@@ -278,22 +278,22 @@ struct CAFFE2_API IValue final {
...
+  // IntListRef
+  IValue(c10::intrusive_ptr<ivalue::IntListRef> v);
   IValue(std::vector<int64_t> v);
   IValue(at::ArrayRef<int64_t> v)
   : IValue(v.vec()) {}
-  bool isIntList() const { return Tag::IntList == tag; }
-  c10::intrusive_ptr<ivalue::IntList> toIntList() && {
-    AT_ASSERT(isIntList());
-    return moveToIntrusivePtr<ivalue::IntList>();
+  bool isIntListRef() const { return Tag::IntListRef == tag; }
```
Comment (smessmer) [inline aten/src/ATen/core/ivalue.h]: This should probably stay `Tag::IntList`

File: aten/src/ATen/core/ivalue.h [hunk 4]
```diff
@@ -278,22 +278,22 @@ struct CAFFE2_API IValue final {
...
-  c10::intrusive_ptr<ivalue::IntList> toIntList() const & {
-    AT_ASSERT(isIntList());
-    return toIntrusivePtr<ivalue::IntList>();
+  c10::intrusive_ptr<ivalue::IntListRef> toIntListRef() const & {
+    AT_ASSERT(isIntListRef());
+    return toIntrusivePtr<ivalue::IntListRef>();
   }
 
-  const std::vector<int64_t>& toIntListRef() const;
+  const std::vector<int64_t>& toIntListRefRef() const;
```
Comment (smessmer) [inline aten/src/ATen/core/ivalue.h]: `RefRef`?

**Review by smessmer — CHANGES_REQUESTED** [review]
Only looked at some of the files, but it seems like the regex replace caught some places you didn't intentionally change. Maybe you can use clang tooling to rename only the `IntList` type instead.

**Comment by apaszke** [comment]
Won't this break C++ extensions people might have?

File: aten/src/ATen/Declarations.cwrap (hunk 1 above)
Comment (gchanan) [reply aten/src/ATen/Declarations.cwrap]: It doesn't really matter, this is just some internal thing to distinguish between sizes and strides for scalar checks.  I'd say `IntListRefSize` is slightly better.

**Comment by gchanan** [comment]
Weren't we going to put a `using` to not break extensions?  It seems okay to ensure we got all the usages first, though.

**Comment by ezyang** [comment]
Declaring bankruptcy on this version of the diff.

**Pull request closed without merging**